tests/lib.rs

```diff
@@ -280,3 +281,91 @@ fn block_height() {
             .unwrap()
     )
 }
+
+#[test]
+fn proof_verifier() {
+    use dusk_plonk::prelude::*;
+    use transfer_circuits::ExecuteCircuit;
+
+    let contract = ProofVerifier::new();
+
+    let store = MS::new();
+
+    let code =
+        include_bytes!("contracts/proof_verification/proof_verification.wasm");
+
+    let contract = Contract::new(contract, code.to_vec(), &store).unwrap();
+
+    let mut network = NetworkState::<StandardABI<_>, MS>::default();
+
+    let contract_id = network.deploy(contract).unwrap();
+
+    let mut gas = GasMeter::with_limit(1_000_000_000);
+
+    let old_crs = rusk_profile::get_common_reference_string();
+    let pp = unsafe {
+        let buff = std::fs::read("tests/pub_params_dev.bin")
+            .expect("Error reading from PubParams file");
+        PublicParameters::from_slice_unchecked(buff.as_slice())
+            .expect("PubParams deser error")
+    };
+
+    rusk_profile::set_common_reference_string("tests/pub_params_dev.bin")
+        .expect("Error setting CRS in rusk_profile");
+    let mut circuit = ExecuteCircuit::<17, 15>::create_dummy_circuit::<_, MS>(
+        &mut rand::thread_rng(),
+        1,
+        1,
+    )
+    .expect("Error creating a dummy setup");
+
+    let (pk, vk) = circuit.compile(&pp).expect("Error compiling the circuit");
+    let proof = circuit
+        .gen_proof(&pp, &pk, b"dusk")
+        .expect("Error generating the proof");
+
+    let mut pi = circuit.get_pi_positions().clone();
+    // Reset PI positions to emulate real-world verification
+    pi.iter_mut().for_each(|p| match p {
+        PublicInput::BlsScalar(_, p) => *p = 0,
+        PublicInput::JubJubScalar(_, p) => *p = 0,
+        PublicInput::AffinePoint(_, p, q) => {
+            *p = 0;
+            *q = 0;
+        }
+    });
+
+    let public_inputs_bytes: Vec<u8> =
+        pi.iter().flat_map(|&inp| inp.to_bytes().to_vec()).collect();
+    let label = String::from("transfer-execute-1-1").into_bytes();
+    let proof_bytes = proof.to_bytes().to_vec();
+    let vk_bytes = vk.to_bytes().to_vec();
+
+    assert!({
+        let mut circuit =
+            ExecuteCircuit::<17, 15>::create_dummy_circuit::<_, MS>(
+                &mut rand::thread_rng(),
+                1,
+                1,
+            )
+            .expect("Error creating a dummy setup");
+        circuit.verify_proof(&pp, &vk, b"dusk", &proof, &pi).is_ok()
+    });
+
+    assert_eq!(
+        true,
+        network
+            .query::<_, bool>(
+                contract_id,
+                (
+                    proof_verification::PROOF_VERIFICATION,
+                    proof_bytes,
+                    vk_bytes,
+                    label,
+                    public_inputs_bytes
+                ),
+                &mut gas
+            )
+            .unwrap()
+    );
+}
```

Comment:
I didn't see setting the crs back, so you're basically override the CRS with the test one.
To be fair, used as is, if the test are failing we won't restore it. What we really need here is a `setup` and `teardown` functions, that it doesn't come out of the box with Rust, but it's not hard to implement; for example this could help: https://medium.com/@ericdreichert/test-setup-and-teardown-in-rust-without-a-framework-ba32d97aa5ab (just one of the many link I get googling)

So, storing the `CRS` will be done in the "setup" part, and restore it in the "teardown" part.

However, if you think is too much to implement now, we can leave the naive implementation, for the time being, but be sure the tests are always passing and you're actually restoring the `old_crs`.